main.nf

```diff
@@ -130,83 +258,127 @@ if( ! nextflow.version.matches(">= 0.30") ){
-	.into { reads_to_trim; reads_fastqc }
-
-// Creating genome1 channels
-Channel
-    .fromPath(params.genome1)
-    .ifEmpty {exit 1, "Cannot find any file for Genome1 matching: ${params.genome1}\n" }
-    .set {genome1rename}
-
-if(params.index1 != '') {
+if(params.readPaths){
+    if(params.singleEnd){
+        Channel
+            .from(params.readPaths)
+            .map { row -> [ row[0], [file(row[1][0])]] }
+            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied" }
+            .into { read_files_fastqc; read_files_trimming }
+    } else {
+        Channel
+            .from(params.readPaths)
+            .map { row -> [ row[0], [file(row[1][0]), file(row[1][1])]] }
+            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied" }
+            .into { read_files_fastqc; read_files_trimming }
+    }
+} else {
     Channel
-        .fromPath(params.index1)
-        .ifEmpty {exit 1, "Cannot find any index matching : ${params.index1}\n"}
-        .set {bt_index_genome1}
+        .fromFilePairs( params.reads, size: params.singleEnd ? 1 : 2 )
+        .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}\nNB: Path needs to be enclosed in quotes!\nIf this is single-end data, please specify --singleEnd on the command line." }
+        .into { read_files_fastqc; read_files_trimming }
 }
 
+/****************
+KRAKEN DB CHANNEL
+*****************/
 
-// Creating genome2 channels
-Channel
-    .fromPath(params.genome2)
-    .ifEmpty {exit 1, "Cannot find any file for Genome2 matching: ${params.genome2}\n" }
-    .set {genome2rename}
+if (params.krakendb.endsWith(".tar.gz")){
+    comp_kraken = file(params.krakendb)
 
-if (params.index2 != '') {
-    Channel
-        .fromPath(params.index2)
-        .ifEmpty {exit 1, "Cannot find any index matching : ${params.index2}\n"}
-        .set {bt_index_genome2}
+    process decomp_kraken {
+        input:
+            file(ckdb) from comp_kraken
+        output:
+            file("kraken") into krakendb_ch
+        script:
+            """
+            tar xvzf $ckdb
+            """
+    }
+} else {
+    krakendb_ch = file(params.krakendb)
 }
 
+/*******************************
+SOURCEPREDICT SOURCES AND LABELS
+********************************/
+sp_labels = file(params.sp_labels, checkIfExists: true)
+sp_sources = file(params.sp_sources, checkIfExists: true)
 
 
-//Logging parameters
+/*******************
+Logging parameters
+********************/
+
 log.info "================================================================"
 log.info " coproID: Coprolite Identification"
 log.info " Homepage / Documentation: https://github.com/maxibor/coproid"
 log.info " Author: Maxime Borry <borry@shh.mpg.de>"
-log.info " Version ${version}"
-log.info " Last updated on ${version_date}"
+log.info " Version ${workflow.manifest.version}"
 log.info "================================================================"
```

Comment:
Can use `log.info nfcoreHeader()` here too?
https://github.com/nf-core/atacseq/blob/c65bf6cd11c8a98807084d2d4039822a19f03969/main.nf#L247

Reply:
I would even advise for that, makes the code a bit cleaner!

Reply:
Fixed with #7